pygmt/src/dimfilter.py

```diff
@@ -0,0 +1,144 @@
+"""
+
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    D="distance",
+    F="filter",
+    G="outgrid",
+    I="spacing",
+    N="sectors",
+    R="region",
+    V="verbose",
+)
+@kwargs_to_strings(I="sequence", R="sequence")
+def dimfilter(grid, **kwargs):
+    r"""
+    Filter a grid by dividing the filter circle.
+
+    Filter a grid in the space (or time) domain by
+    dividing the given filter circle into the given number of sectors,
+    applying one of the selected primary convolution or non-convolution
+    filters to each sector, and choosing the final outcome according to the
+    selected secondary filter. It computes distances using Cartesian or
+    Spherical geometries. The output grid can optionally be generated as a
+    subregion of the input and/or with a new increment using ``spacing``,
+    which may add an "extra space" in the input data to prevent edge
+    effects for the output grid. If the filter is low-pass, then the output
+    may be less frequently sampled than the input. **dimfilter** will not
+    produce a smooth output as other spatial filters
+    do because it returns a minimum median out of *N* medians of *N*
+    sectors. The output can be rough unless the input data is noise-free.
+    Thus, an additional filtering (e.g., Gaussian via :func:`pygmt.grdfilter`)
+    of the DiM-filtered data is generally recommended.
+
+    Full option list at :gmt-docs:`dimfilter.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    distance : int or str
+        Distance flag tells how grid (x,y) relates to filter width, as follows:
+
+        - **0**\ : grid (x,y) in same units as *width*, Cartesian distances.
+        - **1**\ : grid (x,y) in degrees, *width* in kilometers, Cartesian
+          distances.
+        - **2**\ : grid (x,y) in degrees, *width* in km, dx scaled by
+          cos(middle y), Cartesian distances.
+
+        The above options are fastest because they allow weight matrix to be
+        computed only once. The next two options are slower because they
+        recompute weights for each latitude.
+
+        - **3**\ : grid (x,y) in degrees, *width* in km, dx scaled by
+          cosine(y), Cartesian distance calculation.
+        - **4**\ : grid (x,y) in degrees, *width* in km, Spherical distance
+          calculation.
+    filter : str
+        **x**\ *width*\ [**+l**\|\ **u**]
+        Sets the primary filter type. Choose among convolution and
+        non-convolution filters. Append the filter code **x** followed by
```

Comment:
Append doesn't quite make sense in PyGMT, maybe just say 'Use'?
```suggestion
        non-convolution filters. Use the filter code **x** followed by
```